executor/aggfuncs/func_first_row.go

```diff
@@ -243,24 +268,24 @@ func (e *firstRow4String) ResetPartialResult(pr PartialResult) {
 func (e *firstRow4String) UpdatePartialResult(sctx sessionctx.Context, rowsInGroup []chunk.Row, pr PartialResult) (memDelta int64, err error) {
 	p := (*partialResult4FirstRowString)(pr)
 	if p.gotFirstRow {
-		return 0, nil
+		return memDelta, nil
 	}
 	if len(rowsInGroup) > 0 {
 		input, isNull, err := e.args[0].EvalString(sctx, rowsInGroup[0])
 		if err != nil {
-			return 0, err
+			return memDelta, err
 		}
 		p.gotFirstRow, p.isNull, p.val = true, isNull, stringutil.Copy(input)
```

Comment:
stringutil.copy will alloc extral memory for a copy of input string, we should add the len of it to memDelta.